### DISCUSSION TITLE
Setting Quota on Bucket

### DISCUSSION THREAD

**Discussion by hardillb** [minio/minio#15823] [Q&A] [answered]
I'm looking to automate creating buckets using the NodeJS library and I'd like to be able to set the quota on each bucket.

It doesn't look possible using the options passed ot the `makeBucket` function.

There must me an API called by the `mc admin` tool, does anybody have a link to the doc for the api or an example?

**Comment by klauspost** [comment, marked as answer]
`mc admin bucket quota`: https://min.io/docs/minio/linux/reference/minio-mc-admin/mc-admin-bucket-quota.html

Calls here: https://github.com/minio/madmin-go/blob/84dbb5c91e5ea12d60af9ead1ed5c5aba57fb3af/quota-commands.go#L57

We do not provide any guarantee of stability for admin API's, so use at your own risk. We recommend calling `mc`.

**Reply by hardillb** [reply]
Shelling out is not an acceptable solution
